subsys/llext/llext_load.c

```diff
@@ -176,6 +178,24 @@ static int llext_find_tables(struct llext_loader *ldr)
 				sect_mem_idx = LLEXT_MEM_STRTAB;
 			}
 			break;
+		case SHT_PREINIT_ARRAY:
+			sect_type_str = "preinit";
+			sect_mem_idx = LLEXT_MEM_PREINIT;
+			if (shdr->sh_entsize != sizeof(void *) ||
+			    shdr->sh_size % shdr->sh_entsize != 0) {
+				LOG_ERR("Invalid %s array", sect_type_str);
+				return -ENOEXEC;
+			}
+			break;
+		case SHT_INIT_ARRAY:
+			sect_type_str = "init";
+			sect_mem_idx = LLEXT_MEM_INIT;
+			if (shdr->sh_entsize != sizeof(void *) ||
+			    shdr->sh_size % shdr->sh_entsize != 0) {
+				LOG_ERR("Invalid %s array", sect_type_str);
+				return -ENOEXEC;
+			}
+			break;
```

Comment:
I started off on the wrong foot, `find_tables` looked nice since it immediately deals with partitions without all the "merge" stuff, but you are right that it isn't an early requirement. Will refactor to use `map_sections` instead.